runtime/rasdump/dmpsup.c

```diff
@@ -532,10 +539,46 @@ configureDumpAgents(J9JavaVM *vm)
 	/* Process IBM_XE_COE_NAME */
 	mapDumpSettings(vm, agentOpts, &agentNum);
 	
-	/* Process -Xdump command-line options (L..R)*/
+
+	/*
+	 * Process -XX:[+-]HeapDumpOnOutOfMemoryError.
+	 * Set heapDumpIndex to the index of the rightmost option
+	 * and indicate whether enable or disable wins.
+	 */
+	heapDumpIndex = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXNOHEAPDUMPONOOM, NULL);
+	xdumpIndex = FIND_AND_CONSUME_ARG(EXACT_MATCH, VMOPT_XXHEAPDUMPONOOM, NULL);
+	processXXHeapDump = ((xdumpIndex >= 0) || (heapDumpIndex >= 0));
+	if (xdumpIndex > heapDumpIndex) {
+		enableXXHeapDump = TRUE;
+		heapDumpIndex = xdumpIndex;
+	}
+
+	/*
+	 * Process -Xdump command-line options (L..R).
+	 * Treat -XX:[+-]HeapDumpOnOutOfMemoryError as an alias of -Xdump.
+	 */
+
 	xdumpIndex = FIND_ARG_IN_VMARGS_FORWARD(OPTIONAL_LIST_MATCH, VMOPT_XDUMP, NULL);
-	while ( xdumpIndex >= 0 )
+	/* handle the case of no -Xdump options */
```

Comment:
If this were moved after the `while` loop below, we wouldn't need to check `xdumpIndex < 0`; `processXXHeapDump` is sufficient.

Reply:
Very clever idea.  Thanks.